src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/redundantmodifier/InputRedundantModifierFinalUnnamedVariables.java

```diff
@@ -0,0 +1,33 @@
+/*
+RedundantModifier
+tokens = (default)METHOD_DEF, VARIABLE_DEF, ANNOTATION_FIELD_DEF, INTERFACE_DEF, \
+         CTOR_DEF, CLASS_DEF, ENUM_DEF, RESOURCE
+jdkVersion = (default)21
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.modifier.redundantmodifier;
+
+public class InputRedundantModifierFinalUnnamedVariables {
```

Comment:
Please add a test case with a switch statement, with pattern matching, which uses unnamed variables (like `case final Integer _ -> System.out.println("Integer");`).

Thinking back, we may have missed this test case in other PRs.

Reply:
You forgot that we took almost 3 days discussing the problem with javac 21 with this example

Reply:
This example will not compile by javac21 although it is 100% valid according to java grammar (It compiles locally on java version. 22). probably a bug that were solved

Reply:
You did not mention this in our discussion/your examples. You only mentioned record patterns.

Reply:
Compiles fine for me:
```
➜  src javac --version
javac 21.0.3

➜  src cat Test.java 
public class Test {
    void m(Object o) {
        switch (o) {
            case final Integer _ -> System.out.println("Integer");
            case String _ -> System.out.println("String");
            default -> System.out.println("Unknown");
        }
    }
}

➜  src javac --enable-preview --release 21 Test.java
Note: Test.java uses preview features of Java SE 21.
Note: Recompile with -Xlint:preview for details.

```

Reply:
Yes sorry, I thought we had this bug with both examples but yes it works for pattern variables. The bug was in record pattern only. 

Done.